From 4c6fc09f06e68727abaec2bcd703d5e4dd8cc586 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Fri, 19 May 2023 14:58:20 -0400
Subject: [PATCH] Issue 5565 - Add upgrade script for new Rust password storage
 plugins

Description: add upgrade ldif for Rust PBKDF2 password storage plugins

Relates: https://github.com/389ds/389-ds-base/issues/5565

Reviewed by: spichugi(Thanks!)
---
 Makefile.am                                   |  1 +
 .../src/scripts/50rust-pwdstorageplugins.ldif | 52 +++++++++++++++++++
 2 files changed, 53 insertions(+)
 create mode 100644 ldap/admin/src/scripts/50rust-pwdstorageplugins.ldif

diff --git a/Makefile.am b/Makefile.am
index 8f2bd4212..b6ea3fe62 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -997,6 +997,7 @@ update_DATA = ldap/admin/src/scripts/exampleupdate.pl \
 	ldap/admin/src/scripts/50refintprecedence.ldif \
 	ldap/admin/src/scripts/50retroclprecedence.ldif \
 	ldap/admin/src/scripts/50rootdnaccesscontrolplugin.ldif \
+	ldap/admin/src/scripts/50rust-pwdstorageplugins.ldif \
 	ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif \
 	ldap/admin/src/scripts/50cryptpwdstorageplugin.ldif \
 	ldap/admin/src/scripts/50contentsync.ldif \
diff --git a/ldap/admin/src/scripts/50rust-pwdstorageplugins.ldif b/ldap/admin/src/scripts/50rust-pwdstorageplugins.ldif
new file mode 100644
index 000000000..696f409e3
--- /dev/null
+++ b/ldap/admin/src/scripts/50rust-pwdstorageplugins.ldif
@@ -0,0 +1,52 @@
+dn: cn=PBKDF2,cn=Password Storage Schemes,cn=plugins,cn=config
+objectclass: top
+objectclass: nsSlapdPlugin
+cn: PBKDF2
+nsslapd-pluginpath: libpwdchan-plugin
+nsslapd-plugininitfunc: pwdchan_pbkdf2_plugin_init
+nsslapd-plugintype: pwdstoragescheme
+nsslapd-pluginenabled: on
+nsslapd-pluginId: ID
+nsslapd-pluginVersion: PACKAGE_VERSION
+nsslapd-pluginVendor: VENDOR
+nsslapd-pluginDescription: DESC
+
+dn: cn=PBKDF2-SHA1,cn=Password Storage Schemes,cn=plugins,cn=config
+objectclass: top
+objectclass: nsSlapdPlugin
+cn: PBKDF2-SHA1
+nsslapd-pluginpath: libpwdchan-plugin
+nsslapd-plugininitfunc: pwdchan_pbkdf2_sha1_plugin_init
+nsslapd-plugintype: pwdstoragescheme
+nsslapd-pluginenabled: on
+nsslapd-pluginId: ID
+nsslapd-pluginVersion: PACKAGE_VERSION
+nsslapd-pluginVendor: VENDOR
+nsslapd-pluginDescription: DESC
+
+dn: cn=PBKDF2-SHA256,cn=Password Storage Schemes,cn=plugins,cn=config
+objectclass: top
+objectclass: nsSlapdPlugin
+cn: PBKDF2-SHA256
+nsslapd-pluginpath: libpwdchan-plugin
+nsslapd-plugininitfunc: pwdchan_pbkdf2_sha256_plugin_init
+nsslapd-plugintype: pwdstoragescheme
+nsslapd-pluginenabled: on
+nsslapd-pluginId: ID
+nsslapd-pluginVersion: PACKAGE_VERSION
+nsslapd-pluginVendor: VENDOR
+nsslapd-pluginDescription: DESC
+
+dn: cn=PBKDF2-SHA512,cn=Password Storage Schemes,cn=plugins,cn=config
+objectclass: top
+objectclass: nsSlapdPlugin
+cn: PBKDF2-SHA512
+nsslapd-pluginpath: libpwdchan-plugin
+nsslapd-plugininitfunc: pwdchan_pbkdf2_sha512_plugin_init
+nsslapd-plugintype: pwdstoragescheme
+nsslapd-pluginenabled: on
+nsslapd-pluginId: ID
+nsslapd-pluginVersion: PACKAGE_VERSION
+nsslapd-pluginVendor: VENDOR
+nsslapd-pluginDescription: DESC
+
-- 
2.40.1

